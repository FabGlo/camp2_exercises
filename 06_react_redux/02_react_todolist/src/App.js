import React, { Component } from 'react';
import './App.css';


class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      value: "",
      taskList: []
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleSubmit(event){
    event.preventDefault();
    let newList= this.state.taskList;
    newList.push({value:this.state.value, check: false})
    this.setState({taskList:newList, value:""})
  }
  handleChange(event) {
    this.setState({value: event.target.value});
  }
  strikeThrough(index){
    let newTaskList = this.state.taskList;
    if(newTaskList[index].check === false)
    newTaskList[index]={
      value: this.state.taskList[index].value,
      check: true
    }
    else {
      newTaskList[index]={
        value: this.state.taskList[index].value,
        check: false
    }
  }
  this.setState({taskList:newTaskList})
}
deleteTask(index){
  let newTaskList2 = this.state.taskList;
  newTaskList2.splice(index,1);
  this.setState({taskList:newTaskList2})
}
  render() {
    return (
      <div>
        <div className="App">
          <h1 className="App-header">Hello to the #1 todoapp on Apple store</h1>
          <form onSubmit={this.handleSubmit}>
            <label>
              Add a task:
              <input type="text" id="task" value={this.state.value} onChange={this.handleChange}/>
            </label>
            <input type="submit" value="Submit"/>
          </form>

          <div className="App-intro">
            {this.state.taskList.map((task,index)=>(
              <div className={"classCheck" + task.check}>
                <span>{task.value}</span>
                <input type="checkbox" checked={task.check} onChange={() => this.strikeThrough(index)}/>
                {task.check
                ?<input type="submit" value="Delete" onClick={() => this.deleteTask(index)}/>
                :<span/>}
                <img src="https://www.google.fr/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&cad=rja&uact=8&ved=0ahUKEwjCn8aJ9c3YAhUHXRQKHeOTDZAQjRwIBw&url=https%3A%2F%2Fgiphy.com%2Fartists&psig=AOvVaw2zvhZsN8shXh9Nq3EuL5yr&ust=1515691523860234"/>
              </div>
            ))}
          </div>
        </div>
      </div>
    )
  }
}

export default App;
