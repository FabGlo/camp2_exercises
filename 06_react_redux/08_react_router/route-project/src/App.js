import React from 'react'
import Home from "./Home";
import Chat from "./Chat";
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom';

const RouterManagement = () => (
  <Router>
    <div>
      <hr/>
      <Route exact path="/" component={Home}/>
      <Route path="/chat" component={Chat}/>
    </div>
  </Router>
)
export default RouterManagement
