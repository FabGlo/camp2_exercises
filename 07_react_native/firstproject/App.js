import React from 'react';
import About from './components/About.js';
import Search from './components/Search.js';
import {TabNavigator} from 'react-navigation';
import {View} from 'react-native';
import globalStyle from "./style.js";

const Tabs = TabNavigator({
  Search: {screen: Search},
  About: {screen: About}
  },{
    tabBarOptions:{
      showIcon: true,
      showLabel: false,
      indicatorStyle:{
        height: 3,
        backgroundColor: "#FFF"
      },
      style:{
        backgroundColor: globalStyle.color,
      }
    },
  }
)

export default class App extends React.Component {
  render() {
    return (
      <View style={{flex:1}}>
        <Tabs />
      </View>
    );
  }
}
