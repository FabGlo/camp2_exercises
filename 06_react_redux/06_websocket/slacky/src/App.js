import React, { Component } from 'react';
import './App.css';
import Chat from './Chat.js';


class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      username: "",
      isLogged: false,
      text: "",
      textList: [],
      users: []
    };
    this.handleChangeU = this.handleChangeU.bind(this);
    this.handleSubmitU = this.handleSubmitU.bind(this);
    this.handleChangeT = this.handleChangeT.bind(this);
    this.handleSubmitT = this.handleSubmitT.bind(this);
  }

  handleSubmitT(event){
    event.preventDefault();
    let newTextList= this.state.textList
    newTextList.push(this.state.text)
    this.setState({textList: newTextList}, ()=>console.log(this.state.textList[0]))
  }

  handleChangeT(event) {
    this.setState({text: event.target.value});
  }

  handleSubmitU(event){
    this.setState({isLogged: !this.state.isLogged});
  }

  handleChangeU(event) {
    this.setState({username: event.target.value});
  }

  componentDidMount(){
    const nbUsersElem = document.getElementById("nbUsers");
    const ws = new WebSocket("ws://localhost:3500");
    ws.onmessage = (event) => {
      console.log("Message: " + event.data);
      nbUsersElem.innerHTML = event.data;
    };
    window.addEventListener("beforeunload", () => ws.send("CLOSE"));
  }

  render() {
    console.log(this.state.textList)
    return (
      <div className="App">
        <h1>Welcome to Slacky!</h1>
        {this.state.isLogged === false
        ?<form onSubmit={this.handleSubmitU}>
          <label>
            Please, give us your username:
            <input type="text" name="username" value={this.state.username} onChange={this.handleChangeU}/>
          </label>
          <input type="submit" value="Submit" />
        </form>
        : <Chat
            text={this.state.text}
            textList={this.state.textList}
            username={this.state.username}
            onSubmit={this.handleSubmitT}
            onChange={this.handleChangeT}
          />}
          <p onClick={this.usersConnected}>There is <span id="nbUsers">0</span> users connected.</p>
      </div>
    );
  }
}

export default App;
