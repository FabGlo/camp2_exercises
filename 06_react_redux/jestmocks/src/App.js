import React, { Component } from 'react';
import './App.css';
const fetch = require ("node-fetch");
const API_KEY= process.env.REACT_APP_OPEN_WEATHER_KEY;


function loadTemperature(fetch){
  return fetch(
    `https://api.openweathermap.org/data/2.5/weather?q=rennes,fr&APPID=${API_KEY}&units=metric`,
    {method: "GET"}
  )
    .then(res => res.json())
    .then(res => res.main.temp)
  }

class App extends React.Component {
  constructor(props){
    super(props);
    this.state={
      temp:0,
    }
  }
  componentDidMount(){
    loadTemperature(fetch)
    .then((res) => {
      this.setState({
        temp: res,
      });
    },
    (error)=>{
      this.setState({
        error
      })
    })
  }
  render() {
    return (
      <div className="App">
        <p>temperature actually on Rennes is {this.state.temp}</p>
      </div>
    );
  }
}

export default App;
export {loadTemperature}
