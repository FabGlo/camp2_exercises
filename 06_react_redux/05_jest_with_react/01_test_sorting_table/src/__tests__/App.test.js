import React from 'react';
import App from '../App.js';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Wapiti from "wapiti";

beforeAll(() => {
  configure({ adapter: new Adapter() });
})


const fakeProducts =
[
  { "decathlon_id": 11, "title": "Corne chasse 14cm", "price": 9.99 },
  { "decathlon_id": 32, "title": "Basic L print Long Gold Fusion", "price": 9.99 },
]

test('Link renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('row sorted in asc after click on head row', () => {
  const sortedRow = shallow(<App lines={fakeProducts}/>);
  sortedRow.find("th").at(1).simulate("click");
  expect(sortedRow.find("td").at(1).text()).toBe("Basic L print Long Gold Fusion");
});

test('row sorted in asc after click on head row', () => {
  expect.assertions(1);
  return Wapiti.goto("http://localhost:3000")
    .click("th")
    .capture(() => document.querySelector("td").textContent)
    .click("th")
    .capture(() => document.querySelector("td").textContent)
    .run()
    .then(result => {
      expect(result).toEqual(["8380024", "8044622"]);
    });
});
