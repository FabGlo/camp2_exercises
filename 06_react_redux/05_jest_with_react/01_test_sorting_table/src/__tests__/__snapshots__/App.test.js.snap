// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div>
  <h1>
    Products
  </h1>
  <table>
    <thead>
      <tr>
        <th
          onClick={[Function]}
        >
          ID
        </th>
        <th
          onClick={[Function]}
        >
          Title
        </th>
        <th
          onClick={[Function]}
        >
          Price
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
